common/argparse.h

```diff
@@ -0,0 +1,895 @@
+  //
+  // This provides the primary descriptive information for commands used by help
+  // and other diagnostic messages. For subcommands, it also provides the name
+  // used to access the subcommand.
+  struct CommandInfo {
+    // The name of the command. For subcommands, this is also the argument
+    // spelling that accesses this subcommand. It must consist of characters in
+    // the set [-a-zA-Z0-9], and must not start with a `-`.
+    llvm::StringRef name;
+
+    // An optional version string that will be rendered as part of version meta
+    // action by the library. While this library does not impose a machine
+    // parsable structure, users will expect this to be extractable and parsable
+    // in practice.
+    //
+    // Subcommands with an empty version will inherit the first non-empty parent
+    // version.
+    //
+    // Whether a (possibly inherited) version string is non-empty determines
+    // whether this library provides the version-printing meta actions via a
+    // `--version` flag or, if there are subcommands, a `version` subcommand.
+    llvm::StringRef version = "";
+
+    // Optional build information to include when printing the version.
+    llvm::StringRef build_info = "";
+
+    // An optional long-form help string for the command.
+    //
+    // When accessing a command's dedicated help output, this will form the main
+    // prose output at the beginning of the help message. When listing
+    // subcommands, the subcommand's `help` string will be used to describe it
+    // in the list.
+    //
+    // The help meta actions are available regardless of whether this is
+    // provided in order to mechanically describe other aspects of the command.
+    //
+    // This field supports multiple lines and uses the text block handling
+    // described in the top-level `Args` comment.
+    llvm::StringRef help = "";
+
+    // An optional short help string for the command.
+    //
+    // This should be very short, at most one line and ideally fitting within a
+    // narrow column of a line. If left blank, the first paragraph of text (up
+    // to the first blank line) in the `help` string will be used.
+    llvm::StringRef help_short = "";
+
+    // An optional custom block of text to describe the usage of the command.
+    //
+    // The usage should just be a series of lines with possible invocations of
+    // the command summarized as briefly as possible. Ideally, each variation on
+    // a single line. The goal is to show the *structure* of different command
+    // invocations, not to be comprehensive.
+    //
+    // When omitted, the library will generate these based on the parsing logic.
+    // The generated usage is expected to be suitable for the vast majority of
+    // users, but can be customized in cases where necessary.
+    llvm::StringRef usage = "";
+
+    // An optional epilog multi-line block of text appended to the help display
```

Comment:
epilog -> epilogue

From what I can find, [epilog is in the minority, even in the US](https://sapling.ai/usage/epilog-vs-epilogue).

Reply:
Before I read this code I was not familiar with this spelling of epilogue. The close link in documentation isn't valuable to me in part because of Carbon's behavior divergence -- even in review, I'm not referring to Python's argparse documentation, and divergent APIs would make that difficult regardless.

Maybe ask on Discord which spelling others would prefer?